mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -229,6 +229,48 @@ static Optional<Type> convertVecType(sycl::VecType type,
   return convertBodyType("class.sycl::_V1::vec", type.getBody(), converter);
 }
 
+int getAddressSpaceAsInt(mlir::sycl::AccessAddrSpace AddrSpace) {
+  switch (AddrSpace) {
+  case AccessAddrSpace::Private:
+    return 0;
+  case AccessAddrSpace::Global:
+    return 1;
+  case AccessAddrSpace::Constant:
+    return 2;
+  case AccessAddrSpace::Local:
+    return 3;
+  case AccessAddrSpace::ExtIntelGlobalDevice:
+    return 4;
+  case AccessAddrSpace::ExtIntelHost:
+    return 5;
+  case AccessAddrSpace::Generic:
+    return 6;
+  default:
+    llvm_unreachable("Unknown address mode");
+  }
```

Comment:
The value of `access::address_space` doesn't relate to the value in LLVM (it is target dependant)

As we only care about SPIR for now,

```suggestion
int convertToLLVMAddressSpace(mlir::sycl::AccessAddrSpace AddrSpace) {
  switch (AddrSpace) {
  case AccessAddrSpace::Private:
    return 0;
  case AccessAddrSpace::Global:
    return 1;
  case AccessAddrSpace::Constant:
    return 2;
  case AccessAddrSpace::Local:
    return 3;
  case AccessAddrSpace::ExtIntelGlobalDevice:
    return 5;
  case AccessAddrSpace::ExtIntelHost:
    return 6;
  case AccessAddrSpace::Generic:
    return 4;
  default:
    llvm_unreachable("Unknown address mode");
  }
```

Reply:
Removed the function